vulkan: add sdk

wirx6 opened the pull request.

Alexpux commented on mingw-w64-glslang/PKGBUILD:
```diff
@@ -1,11 +1,12 @@
+_gitver=714e58b
 pkgver=3.0.git2.${_gitver}
-pkgrel=1
+pkgrel=0
```
revert pkgrel to 1

Alexpux commented on mingw-w64-glslang/PKGBUILD:
```diff
@@ -18,36 +19,19 @@ sha256sums=('SKIP')
 build() {
   [[ -d "${srcdir}"/build-${CARCH} ]] && rm -rf "${srcdir}"/build-${CARCH}
   mkdir -p "${srcdir}"/build-${CARCH} && cd "${srcdir}"/build-${CARCH}
-
```
Revert this block too.

Alexpux commented on mingw-w64-glslang/PKGBUILD:
```diff
@@ -18,36 +19,19 @@ sha256sums=('SKIP')
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
   ${MINGW_PREFIX}/bin/cmake \
     -G'MSYS Makefiles' \
     -DCMAKE_BUILD_TYPE=Release \
```
remove this option

wirx6 commented:
Done

wirx6 commented:
Is anyone here?

wirx6 commented:
@Alexpux I'm open for new comments but don't know what I haven't fixed.
- changed `pkgrel` back to 1
- restored `extra_config`
- removed conflicting `-DCMAKE_BUILD_TYPE=Release`

If it's about buildbot fail: probably linker error because `InitOnceExecuteOnce` is defined in `kernel32.dll` what is already being linked. I tested both packages and anything is working fine.

Alexpux commented:
See carefully my comments. You not fix ````pkgrel```` and downloading tarball into named file

wirx6 commented on mingw-w64-glslang/PKGBUILD:
```diff
@@ -1,9 +1,10 @@
-_gitver=eee9d53
+_gitver=714e58b
 pkgver=3.0.git2.${_gitver}
 pkgrel=1
```
I'm sure i reverted changes to `pkgrel` as you wished.
Tarball wasn't mentioned before and it's impossible to do because Khronos don't release glslang source tarballs suitable for their other tools.
There is required revision: https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/sdk-1.0.46/external_revisions/glslang_revision

Alexpux commented:
You provide several PKGBUILDs in this pull request.
Look at vulkan PKGBUILD - comments is there

wirx6 commented:
I see only comments to glslang.
![1498113448744](https://user-images.githubusercontent.com/18199294/27420448-30574f4a-5726-11e7-93b9-accbaffa9989.png)

Alexpux commented:
@wirx6  this is what I want from you https://github.com/Alexpux/MINGW-packages/commit/b4a4b3cdf46d0b914faccf3a86012f04718cd102

Alexpux commented:
In any case 32 bit package fail to build

wirx6 commented:
@Alexpux I don't know why it's happening, `InitOnceExecuteOnce` should be defined in `kernel32.dll`
https://github.com/Alexpux/mingw-w64/blob/master/mingw-w64-crt/lib32/kernel32.def#L757
Also https://github.com/KhronosGroup/OpenCL-ICD-Loader gives the same error on 32.

mati865 commented:
MS docs say it is included in Vista or newer `To compile an application that uses this function, define _WIN32_WINNT as 0x0600 or later.`

@wirx6 you added it to CXX flags but notice object in error: `trampoline.c.obj):trampoline.c`, theoretically it should work after you add it to C flags as well.

wirx6 commented:
@mati865 `trampoline.c` is compiling well, problem is occurring only during linking of 32-bit version(no problems with 64-bit version).
Compiler should throw error earlier if any undeclared functions were used.

mati865 commented:
After adding this define to CFLAGS it goes further:
```
[100%] Linking CXX executable vkjson_info.exe
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x125): undefined reference to `vkGetPhysicalDeviceProperties@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x137): undefined reference to `vkGetPhysicalDeviceFeatures@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x149): undefined reference to `vkGetPhysicalDeviceMemoryProperties@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x16d): undefined reference to `vkGetPhysicalDeviceQueueFamilyProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x1c0): undefined reference to `vkGetPhysicalDeviceQueueFamilyProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x1ec): undefined reference to `vkEnumerateDeviceExtensionProperties@16'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x24b): undefined reference to `vkEnumerateDeviceExtensionProperties@16'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x26f): undefined reference to `vkEnumerateDeviceLayerProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x2b2): undefined reference to `vkGetPhysicalDeviceFormatProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x419): undefined reference to `vkEnumerateDeviceLayerProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x522): undefined reference to `vkEnumerateInstanceLayerProperties@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x5ae): undefined reference to `vkEnumerateInstanceLayerProperties@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x646): undefined reference to `vkEnumerateInstanceExtensionProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x69a): undefined reference to `vkEnumerateInstanceExtensionProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x736): undefined reference to `vkCreateInstance@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x767): undefined reference to `vkEnumeratePhysicalDevices@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x788): undefined reference to `vkDestroyInstance@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0x94c): undefined reference to `vkEnumerateInstanceExtensionProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0xa3a): undefined reference to `vkEnumerateInstanceExtensionProperties@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0xc88): undefined reference to `vkEnumeratePhysicalDevices@12'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0xca5): undefined reference to `vkDestroyInstance@8'
libvkjson.a(vkjson_instance.cc.obj):vkjson_instance.cc:(.text+0xfa5): undefined reference to `vkDestroyInstance@8'
collect2.exe: error: ld returned 1 exit status
```

Alexpux commented:
@mati865 maybe also need remove def file from linlink dlls or fix them for 32 bit
